src/controllers/project.controller.ts

```diff
@@ -185,10 +187,19 @@ export class ProjectController {
 		});
 	}
 
-	async selectDeployment(deploymentId: string): Promise<void> {
-		this.selectedDeploymentId = deploymentId;
+	setSessionsStateFilter(filterState: string) {
+		this.filterSessionsState = filterState;
+		this.fetchSessions();
+	}
 
-		const { data: sessions, error } = await SessionsService.listByDeploymentId(deploymentId);
+	async fetchSessions() {
+		if (!this.selectedDeploymentId) {
+			return;
+		}
+		const { data: sessions, error } = await SessionsService.listByDeploymentId(
+			this.selectedDeploymentId,
+			reverseSessionStateConverter(this.filterSessionsState as SessionStateType)
```

Comment:
Can this fail?

Reply:
As it seems, it's written with all validations that it won' be able to fail. Do you think otherwise?

Reply:
Refactored to try-catch